tests/test_functional/test_preparation/test_causal/test_utils.py

```
@@ -0,0 +1,424 @@
+        """Test that the function returns correct type and structure."""
+        # 1. Setup sample data
+        exposures = pd.DataFrame({PID_COL: [1, 2], ABSPOS_COL: [100, 200]})
+        n_hours_compliance = 24
+
+        # 2. Execute function
+        result = get_non_compliance_abspos(exposures, n_hours_compliance)
+
+        # 3. Assertions
+        self.assertIsInstance(result, dict, "Should return a dictionary")
+        self.assertTrue(
+            all(isinstance(k, int) for k in result.keys()),
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)